client/components/table/table.js

```diff
@@ -84,6 +95,17 @@ class Table extends Component {
 		};
 	}
 
+	updateTableShadow = () => {
+		const table = this.container.current;
+		const scrolledToEnd = table.scrollWidth - table.scrollLeft <= table.offsetWidth;
+		this.setState( {
+			shadowStyles: {
+				transform: 'translateX(' + this.container.current.scrollLeft + 'px)',
```

Comment:
Updating CSS positions with JavaScript is a bit problematic. I recorded this in my PC:
![peek 2018-10-17 16-16](https://user-images.githubusercontent.com/3616980/47094736-0e977100-d22c-11e8-8447-590d663b8d17.gif)

It's hard to see but there is one frame in which the scroll has already happened but the shadow position hasn't been updated yet:
![peek 2018-10-17 16-16](https://user-images.githubusercontent.com/3616980/47094771-1fe07d80-d22c-11e8-8f04-05d08240f870.png)

I imagine that will happen more often in low-end phones.

I have been doing tests with this PR and actually that's a tricky CSS issue, the only way I could find to avoid having to update the position with JavaScript was creating a wrapper `div` around the table which would contain the shadow absolutely positioned to the right. Since that `div` would not scroll, the shadow would always be aligned to the right without having to set its position with JavaScript. But that means polluting the markup, so I'm not sure it's the best solution either.

Given that it's a small visual glitch and there isn't any perfect solution, I'm ok with merging this PR as it is now, but wanted to point it out in case somebody has any other idea. :slightly_smiling_face:

Reply:
Ah, I thought this might happen.  

I updated this in https://github.com/woocommerce/wc-admin/pull/546/commits/6e16f258c2de953dc11ef9b170a1839e36ebdd8e to be relative to the container table body which should let us position it without constantly updating the transform.  This required a small change to how the pagination is positioned on tables to go above the shadow, but I think it works well across all tables.

Let me know if that looks better on your end!

Reply:
Right, that was a clever way to fix that. The problem is solved now and it works like a charm!

I have one last suggestion, what would you think about getting rid of `<div className="woocommerce-table__shadow">` and implementing it as a pseudo-element that is visible/hidden depending on a class applied to `woocommerce-table__table`?

I think now it's do-able and this way we will keep the markup simple.

Reply:
That seems like a good idea.  Pushed in https://github.com/woocommerce/wc-admin/pull/546/commits/159fde6428ee2c10fb87b7f5dc2991743a61df2b

Reply:
Looks really good to me, now. :clap: